functions/registries.go

```diff
@@ -0,0 +1,185 @@
+package functions
+
+import (
+	"github.com/substrait-io/substrait-go/extensions"
+	"github.com/substrait-io/substrait-go/types"
+	"strings"
+)
+
+type TypeRegistry interface {
+	// GetTypeFromTypeString Given a Substrait standard typeString, get the Substrait type. Return
+	// error if type doesn't parse. This type registry should support both standard types and any
+	// available extension types.
+	GetTypeFromTypeString(typeString string) (types.Type, error)
+
+	//GetTypeClasses() []types.TypeClass // TODO define TypeClass so we can get things like DECIMAL<?,?>.
+}
+
+// LocalTypeRegistry is a registry that contains all types associated with a particular dialect.
+type LocalTypeRegistry interface {
+	// GetTypeFromTypeString Given a Substrait standard typeString, get the Substrait type.
+	GetTypeFromTypeString(typeString string) (types.Type, error)
```

Comment:
I don't understand this comment. Local Type Registry is a localized version of a type registry. The TypeRegistry should not be exposed since it is non-localized.